crates/router/src/core/customers.rs

```diff
@@ -34,23 +40,69 @@ pub async fn create_customer(
     let merchant_id = &merchant_account.merchant_id;
     customer_data.merchant_id = merchant_id.to_owned();
 
+    let key = get_key_and_algo(db, merchant_id.clone())
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
     if let Some(addr) = &customer_data.address {
         let customer_address: api_models::payments::AddressDetails = addr
             .peek()
             .clone()
             .parse_value("AddressDetails")
             .change_context(errors::ApiErrorResponse::AddressNotFound)?;
+
         db.insert_address(domain::address::Address {
             city: customer_address.city,
             country: customer_address.country,
-            line1: customer_address.line1,
-            line2: customer_address.line2,
-            line3: customer_address.line3,
-            zip: customer_address.zip,
-            state: customer_address.state,
-            first_name: customer_address.first_name,
-            last_name: customer_address.last_name,
-            phone_number: customer_data.phone.clone(),
+            line1: customer_address
+                .line1
+                .async_map(|inner| crypto::Encryptable::encrypt(inner, &key, GcmAes256 {}))
```

Comment:
create a closure called encrypt = |inner| crypto::Encryptable::encrypt(inner, &key, GcmAes256 {});

so we can do: 
 ``` 
   customer_address
     .line1
     .async_map(encrypt)
```

Reply:
cool bro 👍